tests/functional_tests/test_container_no_root.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+
+import pytest
+import testinfra
+
+from tests.functional_tests.conftest import get_pod_running_containers
+
+container_ignore_list = ["astronomer-kubed"]
+
+
+@pytest.mark.parametrize(
+    "container_info",
+    get_pod_running_containers().values(),
+    ids=get_pod_running_containers().keys(),
```

Comment:
Workaround to get proper ids here. Example build: https://app.circleci.com/pipelines/github/astronomer/astronomer/4273/workflows/2d5e8308-5408-4d25-be48-aa9383635917/jobs/28405

Reply:
The way this is written, it calls `get_pod_running_containers()` twice, which causes the kubernetes client to query kube api for the list of pods twice, and the result may be different. It would be more efficient and more accurate to query kube api once and generate two lists from that one query result. We could try to get creative, but i think the easiest way to do this is to just make the variables outside the function call.